miniwallet/lib/utils.ts

```diff
@@ -0,0 +1,62 @@
+import config from '../config'
+import { ethers } from 'hardhat'
+import { promises as fs, constants as fsConstants, existsSync, mkdirSync } from 'fs'
+import path from 'path'
+
+export const BASE_TEN = 10
+
+export async function checkDeployed (hre, contract) {
+  // If we don't save deployments (e.g. for local networks) deployed = fals
+  if (!hre.network.saveDeployments) { return false }
+  const artifactsDirectory = config.artifactsDirectory + hre.network.name + '/'
+  const artifactName = 'MiniWallet'
+  const artifactFile = artifactsDirectory + artifactName + '.json'
+  try {
+    const currentDeployment = await fs.readFile(artifactFile, { encoding: 'utf-8' })
+    console.log(`Have already deployed ${artifactName}`)
+    console.log(`Current Deployment: ${JSON.stringify(currentDeployment)}`)
+    return true
+  } catch (ex) {
+    // console.log(ex)
+    // If we have an exception assume the artifact hasn't been deployed
```

Comment:
That's dangerous - check for specific exception?

Reply:
Done 🙏 
Notes [here](https://github.com/polymorpher/sms-wallet/blob/ws-miniwallet-v0.2/miniwallet/devlog/PROXY.md#approach)
* `checkDeployed` : Checks if an existing contract has been deployed. The assumption is that once deployed we should not deploy again unless we are doing an upgrade via the proxy. It handles the conditions with the following return
  * true: If this version of the contract (verified by bytecodeHash) has been deployed
  * false: If a different version of the contract (verified by byteCodeHash) has been deployed
  * false: If the contract has not been deployed (no deployment file exists)
  * exits: with any other exception when reading the deployment artifacts
* `persistDeployment`: Persists the contract and proxy information for a contract.